src/apiserver/db_models.py

```diff
@@ -0,0 +1,50 @@
+import uuid
+from enum import Enum
+import time
+import typing
+from pydantic import BaseModel
+
+class RunPhase(Enum):
+    """
+    # QUEUED
+    Run enters this state when received. 
+    In this state run can be returned to invoker for judging.
+    # LOCKED
+    Some invoker is judging this run right now.
+    There are two possibilities:
+    1) Judging finished, and run transitions to the FINISHED state.
+    2) Timeout is hit, and run transitions to the QUEUED state. (TBD)
+    # FINISHED
+    Judging is finished. All files and statuses are uploaded
+    """
+    QUEUED = "queued"
+    LOCKED = "locked"
+    FINISHED = "finished"
+
+class RunMainProj(BaseModel):
+    id: uuid.UUID
+    toolchain_id: str
+    problem_id: str
+    user_id: uuid.UUID
+    contest_id: str
+    phase: str  # RunPhase
+    statuses: typing.List[typing.Tuple[str, str]] = []
+    
+    """
+    Each item is (kind, code) as in invoker_api::Status
+    """
+
+RunMainProj.FIELDS = ['id', 'toolchain_id', 'problem_id', 'user_id', 'contest_id', 'statuses']
+ 
+class RunSourceProj(BaseModel):
+    source: typing.Optional[bytes]
+
+class RunBinaryProj(BaseModel):
+    binary: typing.Optional[bytes]
+    
+class RunProtocolsProj(BaseModel):
+    protocols: typing.Mapping[str, str] = dict()
```

Comment:
This would normally make all instances of RunProtocolsProj share the same mapping. This does not seem to be the intended behavior. Please add a comment with clarification.

Reply:
f**k

Как сделать так, чтобы каждый объект получал собственный пустой словарь, без написания конструктора?

Reply:
Если это не какая-то openapi-шная магия то никак.